go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go

```diff
@@ -280,3 +219,65 @@ func TestDMLOnNewTable(t *testing.T) {
 	defer utils.Exec(t, conn, `delete from t8`)
 	utils.AssertMatchesNoOrder(t, conn, `select id from new_table_tracked join t8`, `[[INT64(0)] [INT64(1)]]`)
 }
+
+// TestNewView validates that view tracking works as expected.
+func TestNewView(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	// insert some data
+	_ = utils.Exec(t, conn, "insert into t2 (id3, id4) values (1, 10), (2, 20), (3, 30)")
+	defer utils.Exec(t, conn, "delete from t2")
+
+	selQuery := "select sum(id4) from t2 where id4 > 10"
+
+	// create a view
+	_ = utils.Exec(t, conn, "create view test_view as "+selQuery)
+	time.Sleep(2 * time.Second)
```

Comment:
This vtgate test is already a little flaky in my experience — especially with GH Actions is otherwise struggling/slow. Using sleeps like this is likely to make it more so. What is the point of this sleep? If we're waiting for some condition to be true, we should check and wait for it with a timeout. IMO these [new] `waitForX()` functions should replace all of the sleeps.

Reply:
Agreed with @mattlord, we could run a query that selects the view in a loop with some sort of timeout. `utils.AssertMatchesWithTimeout` allows us to do that.

Reply:
handled.